duration.go

```diff
@@ -122,33 +137,101 @@ func Parse(d string) (*Duration, error) {
 	return duration, nil
 }
 
-// ToTimeDuration converts the *Duration to the standard library's time.Duration
-// note that for *Duration's with period values of a month or year that the duration becomes a bit fuzzy
+// FromTimeDuration converts the given time.Duration into duration.Duration.
+// Note that for *Duration's with period values of a month or year that the duration becomes a bit fuzzy
+// since obviously those things vary month to month and year to year
+func FromTimeDuration(d time.Duration) (*Duration, error) {
+	if d < 0 {
+		return nil, ErrUnexpectedNegative
+	}
+
+	duration := &Duration{}
+	if d == 0 {
+		return duration, nil
+	}
+
+	if d.Hours() >= hoursPerYear {
+		duration.Years = math.Floor(d.Hours() / hoursPerYear)
+		d -= time.Duration(duration.Years) * nsPerYear
+	}
+	if d.Hours() >= hoursPerMonth {
+		duration.Months = math.Floor(d.Hours() / hoursPerMonth)
+		d -= time.Duration(duration.Months) * nsPerMonth
+	}
+	if d.Hours() >= hoursPerWeek {
+		duration.Weeks = math.Floor(d.Hours() / hoursPerWeek)
+		d -= time.Duration(duration.Weeks) * nsPerWeek
+	}
+	if d.Hours() >= hoursPerDay {
+		duration.Days = math.Floor(d.Hours() / hoursPerDay)
+		d -= time.Duration(duration.Days) * nsPerDay
+	}
+	if d.Hours() >= 1 {
+		duration.Hours = math.Floor(d.Hours())
+		d -= time.Duration(duration.Hours) * nsPerHour
+	}
+	if d.Minutes() >= 1 {
+		duration.Minutes = math.Floor(d.Minutes())
+		d -= time.Duration(duration.Minutes) * nsPerMinute
+	}
+	duration.Seconds = d.Seconds()
+	return duration, nil
+}
+
+// Format formats the given time.Duration into an ISO 8601 duration string (e.g. P1DT6H5M),
+// negative durations are prefixed with a minus sign, for a zero duration "PT0S" is returned.
+// Note that for *Duration's with period values of a month or year that the duration becomes a bit fuzzy
+// since obviously those things vary month to month and year to year
+func Format(d time.Duration) string {
```

Comment:
We should drop this `Format` function entirely. If support for negatives is desired then we should add that to the `String` function.